ash-window/src/lib.rs

```diff
@@ -14,9 +14,15 @@ use ash::extensions::ext; // portability extensions
 ///
 /// # Safety
 ///
-/// In order for the created [`vk::SurfaceKHR`] to be valid for the duration of its
-/// usage, the [`Instance`] this was called on must be dropped later than the
-/// resulting [`vk::SurfaceKHR`].
+/// The resulting [`vk::SurfaceKHR`] **must** be destroyed through
+/// [`khr::Surface::destroy_surface()`] (_not_ [dropped][drop()]) before the [`Instance`] and
+/// [`Entry`] this was called on is [destroyed][Instance::destroy_instance()], in the broad sense
+/// of the word. That is, valid Vulkan usage requires the [`vk::SurfaceKHR`] to be destroyed, and
+/// it otherwise becomes invalid to use after the [`Instance`] was
+/// [destroyed][Instance::destroy_instance()] and/or the [`Entry`] was [dropped][drop()].
```

Comment:
I'm not sure what "in the broad sense of the word" is referring to here. Editing detritus?

> valid Vulkan usage requires the [`vk::SurfaceKHR`] to be destroyed

Not strictly true, I think? You can kill the process without destroying anything and that's fine from a correctness standpoint.

Reply:
> I'm not sure what "in the broad sense of the word" is referring to here. Editing detritus?

To `**must**`, but that link fell away after more words got inserted. Seems to be called editing detritus :)

Reply:
> Not strictly true, I think? You can kill the process without destroying anything and that's fine from a correctness standpoint.

That's one possible edge-case to consider. Perhaps we should reword it using the parent/child relation quoted above, as long as `when there is a parent/child relationship` is converted into **this is a parent/child relationship**, clearly stating the inter-dependency on all the types involved in this PR.

Reply:
SGTM.

Reply:
Almost a year (just shy of a week) in the making, I've now reworded this to say that _if_ you were about to destroy one of the parent objects (detailing which, of course), the child object should be destroyed first.